src/main/scala/mesosphere/marathon/state/RootGroup.scala

```diff
@@ -427,13 +440,14 @@ object RootGroup {
     pods: Map[AbsolutePathId, PodDefinition] = Group.defaultPods,
     groupsById: Map[AbsolutePathId, Group] = Group.defaultGroups,
     dependencies: Set[AbsolutePathId] = Group.defaultDependencies,
-    version: Timestamp = Group.defaultVersion): RootGroup = new RootGroup(apps, pods, groupsById, dependencies, version)
+    autoCreateStrategy: NewGroupStrategy = NewGroupStrategy.Fail,
```

Comment:
slightly confused what is autocreate here, should it be newGroupStrategy?

Reply:
also why is it fail by default? 🤔

Reply:
I renamed it to newGroupStrategy, as that name is used in other places as well.

Reply:
Thank you!